chef/cookbooks/nova/recipes/compute.rb

```diff
@@ -156,47 +187,27 @@ def set_boot_kernel_and_trigger_reboot(flavor='default')
       end
   end
 
-
   # change libvirt to run qemu as user qemu
-  unless %w(redhat centos).include?(node.platform)
-    ruby_block "change qemu user used by libvirt" do
-      block do
-        rc = Chef::Util::FileEdit.new("/etc/libvirt/qemu.conf")
-
-        # make sure to only set qemu:kvm for kvm and qemu deployments, use
-        # system defaults for xen
-        if ['kvm','qemu'].include?(node[:nova][:libvirt_type])
-          rc.search_file_replace_line(/^[ #]*user *=/, 'user = "qemu"')
-          rc.search_file_replace_line(/^[ #]*group *=/, 'group = "kvm"')
-        else
-          rc.search_file_replace_line(/^ *user *=/, '#user = "root"')
-          rc.search_file_replace_line(/^ *group *=/, '#group = "root"')
-        end
-
-        if rc.file_edited?
-          rc.write_file
-          # manually restart libvirtd; we can't do that with a notification to the libvirtd service because we're in a ruby_block
-          %x{rclibvirtd restart}
-        end
-      end
-    end
+  # make sure to only set qemu:kvm for kvm and qemu deployments, use
+  # system defaults for xen
+  if ['kvm','qemu'].include?(node[:nova][:libvirt_type])
+    libvirt_user = "qemu"
+    libvirt_group = "kvm"
   else
-    if ['kvm','qemu'].include?(node[:nova][:libvirt_type])
-      libvirt_user = "qemu"
-      libvirt_group = "kvm"
-    else
-      libvirt_user = "root"
-      libvirt_group = "root"
-    end
+    libvirt_user = "root"
+    libvirt_group = "root"
+  end
 
-    bash "edit qemu config" do
-      only_if "cat /etc/libvirt/qemu.conf | grep 'user =' | grep -q -v '#{libvirt_user}' || cat /etc/libvirt/qemu.conf | grep 'group =' | grep -q -v '#{libvirt_group}'"
-      code <<-EOH
-       sed -i 's|user *=.*|user = "#{libvirt_user}"|g' /etc/libvirt/qemu.conf
-       sed -i 's|group *=.*|group = "#{libvirt_group}"|g' /etc/libvirt/qemu.conf
-      EOH
-      notifies :restart, "service[libvirtd]"
-    end
+  template "/etc/libvirt/qemu.conf" do
+    source "qemu.conf.sle12.erb" if node.platform == "suse" && node.platform_version.to_f >= 12.0
```

Comment:
Me too. But I disliked the other solution even more. For the reasons outlined in https://github.com/crowbar/barclamp-nova/pull/400#discussion_r22035556